msu/systems/serialization/serialization_types/abstract_data.nut

```diff
@@ -0,0 +1,31 @@
+::MSU.Class.AbstractData <- class
+{
+	__Type = null;
+	__Data = null;
+
+	constructor( _type, _data )
+	{
+		this.__Type = _type;
+		this.__Data = _data;
+	}
+
+	function getType()
+	{
+		return this.__Type;
+	}
+
+	function getData()
+	{
+		return this.__Data;
+	}
+
+	function serialize( _out )
+	{
+		_out.writeU8(this.__Type);
+	}
+
+	// Must be overridden by children
```

Comment:
should this throw?

Reply:
No one should be instantiating this class. We could add a throw just to make it absolutely clear LOL but idk if that is necessary.